app_dart/lib/src/request_handlers/scheduler/scheduler_request_subscription.dart

```diff
@@ -59,38 +61,90 @@ class SchedulerRequestSubscription extends SubscriptionHandler {
     ///
     /// Log error message when still failing after retry. Avoid endless rescheduling
     /// by acking the pub/sub message without throwing an exception.
-    String? unscheduledBuilds;
+    ///
+    /// Inform check_runs of failures so they are not infinitely scheduled.
+    final errors = <String>[];
+    var unscheduledWarning = '';
     try {
       await retryOptions.retry(
         () async {
-          final List<bbv2.BatchRequest_Request> requestsToRetry = await _sendBatchRequest(batchRequest);
+          final requestsToRetry = await _sendBatchRequest(batchRequest);
 
           // Make a copy of the requests that are passed in as if simply access the list
           // we make changes for all instances.
-          final List<bbv2.BatchRequest_Request> requestListCopy = [];
-          requestListCopy.addAll(requestsToRetry);
+          final requestListCopy = [...requestsToRetry.retries];
```

Comment:
It's not clear what `requestListCopy` does here, I think we could just use `requestsToRetry.retries` on line 77 and 81 (unless something mutates it between those two, but that is terrifying)

Reply:
This is just compressing the previous code's `addAll()`. I can just back it out if you want; but `_sendBatchRequest`  does mutate `batchRequest.requests`, which is done in a loop - and this does get called multiple times.. e.g.

1. Schedule 5 tests
2. 3 schedule 2 fail - return 2
3. loop: schedule 2 tests

Honestly; having written the above; my head hurts. The proto that is mutated would just be passed right back in after having the elements removed. 

I test on Monday and see what is really going on.

Reply:
I actually changed this to the function creating its own list rather than mutating the containing list. Take another look?